### PR TITLE
move initsystem to kubeadm

### PR DESCRIPTION
**Pull request by yastij**
Signed-off-by: Yassine TIJANI <ytijani@vmware.com>

**What type of PR is this?**

/kind cleanup
/priority important-soon

**What this PR does / why we need it**: ref kubernetes/kubeadm#1600

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

/cc @rosti 
/assign @neolit123 

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

**Review by neolit123 — COMMENTED** [review]
thanks @yastij 
i'm OK with the new location.

A) kubeadm is the only consumer
B) futher decouples kubeadm from /pkg

LGTM
/approve
/priority backlog
/kind cleanup
/assign @liggitt @timothysc 
@kubernetes/sig-cluster-lifecycle-pr-reviews

**Comment by liggitt** [comment]
/approve

thanks!

**Review by rosti — APPROVED** [review]
Thanks @yastij !
Makes perfect sense to me as kubeadm is the only user of it.

**Comment by yastij** [comment]
@rosti - fixed golint failures, some comments were missing, added them

**Review by neolit123 — COMMENTED** [review]
/lgtm

File: cmd/kubeadm/app/util/initsystem/initsystem.go [hunk 1]
```diff
@@ -16,6 +16,7 @@ limitations under the License.
 
 package initsystem
 
+// InitSystem is the interface that describe behaviors of an init system
```
Comment (timothysc) [inline cmd/kubeadm/app/util/initsystem/initsystem.go]: So we are the only place that used this library?

Comment (yastij) [reply cmd/kubeadm/app/util/initsystem/initsystem.go]: Yes

**Pull request merged**